pkg/dependencies/helmfactory.go

```diff
@@ -0,0 +1,138 @@
+package dependencies
+
+import (
+	"context"
+	"sync"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// ExecutableBuilder builds the Helm executeble and returns it.
+type ExecutableBuilder interface {
+	BuildHelmExecutable(...executables.HelmOpt) *executables.Helm
+}
+
+// HelmFactoryOpt configures a HelmFactory.
+type HelmFactoryOpt func(*HelmFactory)
+
+// HelmFactory is responsible for creating and owning instances of Helm client.
+type HelmFactory struct {
+	client         cluster.Client
+	helmClient     *cluster.HelmClient
+	mu             sync.Mutex
```

Comment:
it doesn't seem like you are using this

Reply:
Ah, yes. My bad, forgot to re-include the use of these after some refactoring.